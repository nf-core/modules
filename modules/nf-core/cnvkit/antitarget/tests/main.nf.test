nextflow_process {

    name "Test Process CNVKIT_ANTITARGET"
    script "../main.nf"
    process "CNVKIT_ANTITARGET"

    tag "modules"
    tag "modules_nfcore"
    tag "cnvkit"
    tag "cnvkit/antitarget"

    test("human - bed") {

        when {
            process {
                """
                input[0] = [
                    [id: 'test'],
                    file(params.modules_testdata_base_path + 'genomics/homo_sapiens/genome/chr21/sequence/multi_intervals.bed', checkIfExists: true)
                ]
                """
            }
        }

        then {
            assertAll(
                { assert process.success },
                { assert snapshot(process.out.versions).match("version") },
                { assert snapshot(file(process.out.bed.get(0).get(1)).readLines()[0..5]).match() }
            )
        }

    }

    test("human - bed - stub") {

        options "-stub"

        when {
            process {
                """
                input[0] = [
                    [id: 'test'],
                    file(params.modules_testdata_base_path + 'genomics/homo_sapiens/genome/chr21/sequence/multi_intervals.bed', checkIfExists: true)
                ]
                """
            }
        }

        then {
            assertAll(
                { assert process.success },
                { assert snapshot(process.out).match() }
            )
        }

    }

}
