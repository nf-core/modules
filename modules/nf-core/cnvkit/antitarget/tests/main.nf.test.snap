{
    "human - bed - stub": {
        "content": [
            {
                "0": [
                    [
                        {
                            "id": "test"
                        },
                        "test.antitarget.bed:md5,3d4d20f9f23b39970865d29ef239d20b"
                    ]
                ],
                "1": [
                    "versions.yml:md5,b5e73ea85743cedc68ca6ef8006e5030"
                ],
                "bed": [
                    [
                        {
                            "id": "test"
                        },
                        "test.antitarget.bed:md5,3d4d20f9f23b39970865d29ef239d20b"
                    ]
                ],
                "versions": [
                    "versions.yml:md5,b5e73ea85743cedc68ca6ef8006e5030"
                ]
            }
        ],
        "meta": {
            "nf-test": "0.8.4",
            "nextflow": "23.10.0"
        },
        "timestamp": "2024-05-13T13:58:45.758206"
    },
    "human - bed": {
        "content": [
            [
                "chr21\t23354500\t23509999\tAntitarget",
                "chr21\t23509999\t23665499\tAntitarget",
                "chr21\t23665499\t23820999\tAntitarget",
                "chr21\t23820999\t23976499\tAntitarget",
                "chr21\t23976499\t24131999\tAntitarget",
                "chr21\t24911498\t25066997\tAntitarget"
            ]
        ],
        "meta": {
            "nf-test": "0.8.4",
            "nextflow": "23.10.0"
        },
        "timestamp": "2024-05-13T13:58:33.088665"
    },
    "version": {
        "content": [
            [
                "versions.yml:md5,b5e73ea85743cedc68ca6ef8006e5030"
            ]
        ],
        "meta": {
            "nf-test": "0.8.4",
            "nextflow": "23.10.0"
        },
        "timestamp": "2024-05-13T13:58:33.070317"
    }
}